crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -957,7 +957,23 @@ impl<'a> SuiTestAdapter<'a> {
 
     fn graphql_variables(&self) -> BTreeMap<String, GraphqlQueryVariable> {
         let mut variables = BTreeMap::new();
-        for (name, addr) in &self.compiled_state.named_address_mapping {
+        let named_addrs: Vec<_> = self
+            .compiled_state
+            .named_address_mapping
+            .iter()
+            .map(|(name, addr)| (name.clone(), addr.to_string()))
+            .collect();
+
+        let objects: Vec<_> = self
+            .object_enumeration
+            .iter()
+            .filter_map(|(oid, fid)| match fid {
+                FakeID::Known(_) => None,
+                FakeID::Enumerated(x, y) => Some((format!("obj_{x}_{y}"), oid.to_string())),
+            })
+            .collect();
+
+        for (name, addr) in named_addrs.iter().chain(objects.iter()) {
```

Comment:
nit: the `collect`s before the `chain` (and the `iter`s to undo them) should not be necessary here.

```suggestion
        let named_addrs = self
            .compiled_state
            .named_address_mapping
            .iter()
            .map(|(name, addr)| (name.clone(), addr.to_string()));

        let objects = self
            .object_enumeration
            .iter()
            .filter_map(|(oid, fid)| match fid {
                FakeID::Known(_) => None,
                FakeID::Enumerated(x, y) => Some((format!("obj_{x}_{y}"), oid.to_string())),
            });

        for (name, addr) in named_addrs.chain(objects) {
```

Reply:
Good catch, fixed.